low-level/files/opt/seagate/sspl/setup/sspl_setup_init.py

```diff
@@ -0,0 +1,195 @@
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+#################################################################
+# This script performs following operations.
+# - Creates datapath as defined in /etc/sspl.conf
+# - Check dependencies for consts.roles other than '<product>'
+#################################################################
+
+import os
+import pwd
+import errno
+
+from cortx.sspl.bin.sspl_constants import file_store_config_path, roles
+from cortx.utils.validator.v_service import ServiceV
+from cortx.utils.validator.v_pkg import PkgV
+from cortx.sspl.lowlevel.files.opt.seagate.sspl.setup.error import SetupError
+
+class SSPLInit:
+    """Init Setup Interface"""
+
+    name = "init"
+
+    SSU_DEPENDENCY_RPMS = [
+                "sg3_utils",
+                "gemhpi",
+                "pull_sea_logs",
+                "python-hpi",
+                "zabbix-agent-lib",
+                "zabbix-api-gescheit",
+                "zabbix-xrtx-lib",
+                "python-openhpi-baselib",
+                "zabbix-collector",
+    ]
+
+    SSU_REQUIRED_PROCESSES = [
+                    "openhpid",
+                    "dcs-collectord",
+    ]
+
+    VM_DEPENDENCY_RPMS = []
+
+    HPI_PATH = '/tmp/dcs/hpi'
+    MDADM_PATH = '/etc/mdadm.conf'
+
+    def __init__(self, args : list):
+        """ init methond for SSPL Setup Init Class"""
+        self.args = args
+        self.role = None
+        self.dp = False
+
+    def check_dependencies(self, role : str):
+        # Check for dependency rpms and required processes active state based
+        # on role
+        if role == "ssu":
+            try:
+                PkgV().validate("rpms", self.SSU_DEPENDENCY_RPMS)
+                ServiceV().validate("isrunning", self.SSU_REQUIRED_PROCESSES)
+            except Exception:
+                raise
+        elif role == "vm" or role == "gw" or role == "cmu":
+            try:
+                # No dependency currently. Keeping this section as it may be
+                # needed in future.
+                PkgV().validate("isrunning", self.VM_DEPENDENCY_RPMS)
+                # No processes to check in vm env
+            except Exception:
+                raise
+
+    def get_uid(self, user_name : str) -> int:
+        uid = -1
+        try :
+            uid =  pwd.getpwnam(user_name).pw_uid
+        except KeyError :
+            pass
+        return uid
+
+    def validate_args(self):
+        if not self.args:
+            raise SetupError(
+                        errno.EINVAL,
+                        "No arguments to init call.\n \
+                        expected options : [-dp] [-r <ssu|gw|cmu|vm|cortx>]]"
+                        )
+        i = 0
+        while i < len(self.args):
```

Comment:
args validation for -dp, -r will not apply anymore, as from conf stor, config would be read using full keys. @mariyappanp  please take care of this